darts/tests/test_regression_models.py

```diff
@@ -1,100 +1,193 @@
 import numpy as np
 import pandas as pd
 
+from .. import TimeSeries
+from ..metrics import rmse
+from ..models import RandomForest
+from ..models import LinearRegressionModel
 from .base_test_class import DartsBaseTestClass
 from ..utils import timeseries_generation as tg
-from ..metrics import r2_score
-from ..models import StandardRegressionModel
 
 
 def train_test_split(features, target, split_ts):
     """
     Splits all provided TimeSeries instances into train and test sets according to the provided timestamp.
 
-    :param features: Feature TimeSeries instances to be split.
-    :param target: Target TimeSeries instance to be split.
-    :return: 4-tuple of the form (train_features, train_target, test_features, test_target)
+    Parameters
+    ----------
+    features : TimeSeries
+        Feature TimeSeries instances to be split.
+    target : TimeSeries
+        Target TimeSeries instance to be split.
+    split_ts : TimeStamp
+        Time stamp indicating split point.
+
+    Returns
+    -------
+    TYPE
+        4-tuple of the form (train_features, train_target, test_features, test_target)
     """
-
-    # split features
-    train_features = []
-    test_features = []
-    for feature in features:
-        train_feature, test_feature = feature.split_after(split_ts)
-        train_features.append(train_feature)
-        test_features.append(test_feature)
-
-    # split target
+    train_features, test_features = features.split_after(split_ts)
     train_target, test_target = target.split_after(split_ts)
 
     return (train_features, train_target, test_features, test_target)
 
 
-def test_models_accuracy(test_case, models, features, target, min_r2):
-    # for every model, test whether it predicts the target with a minimum r2 score of `min_r2`
-    train_f, train_t, test_f, test_t = train_test_split(features, target, pd.Timestamp('20010101'))
-
-    for model in models:
-        model.fit(train_f, train_t)
-        prediction = model.predict(test_f)
-        current_r2 = r2_score(prediction, test_t)
-        test_case.assertTrue(current_r2 >= min_r2, "{} model was not able to denoise data."
-                             "A r2 score of {} was recorded.".format(str(model), current_r2))
-
-
 class RegressionModelsTestCase(DartsBaseTestClass):
```

Comment:
Could you perhaps add one or two test cases with some series containing NaNs in order to test this potential issue here?

Reply:
Do you mean the issue discussed before when dropping rows? For that I now included tests so that only the first rows are dropped for which not enough lags are available. 
Or do you mean some performance tests when using nans?

Reply:
I meant the issue discussed before when dropping rows, so I guess we are good.